specification/under_development/Base-Profile.md

```diff
@@ -0,0 +1,160 @@
+# Base Profile
+
+This profile defines a subset of the QIR specification to support a coherent set of functionalities and capabilities that might be offered by a system. The Base Profile specifies the minimal requirements for executing quantum programs. Execution of a Base Profile compliant program does, for instance, not require support executing quantum instructions dependent on measurement outcomes. The only requirements for a backend are to be able to perform unitary transformations on the quantum state, as well as measurements at the end of the program.
+
+TODO: table with list of features that need to be supported by hardware, and what it doesn't need to support?  
+
+To support execution of Base Profile compliant programs, a QPU must have the following fundamental capabilities:
```

Comment:
I think we could make good use of (and refer to) [RFC2119](https://www.ietf.org/rfc/rfc2119.txt).

Or something similar written on our own, for our specific needs, that would serve as a basis to any Profile specification, so that they all use the same vocabulary with the same meaning.

Reply:
That may be useful. The Joint Development Foundation also has some guidance for writing specifications, that I meant to read more carefully...

Reply:
@LaurentAjdnik Would you mind taking a look at the latest update and suggest edits where you think these could be useful but aren't currently used?